sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeMapData.java

```diff
@@ -65,6 +65,22 @@ public UnsafeArrayData valueArray() {
   }
 
   @Override
+  public int hashCode() {
+    int h = numElements;
+    return (h * 31 + keys.hashCode()) * 31 + values.hashCode();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof UnsafeMapData) {
+      UnsafeMapData map = (UnsafeMapData) obj;
+      return numElements == map.numElements && keys.equals(map.keyArray())
+        && values.equals(map.valueArray());
```

Comment:
how about elements orders? `Map(1 -> "a", 2 -> "b")` should be equal to `Map(2 -> "b", 1 -> "a")`